pkg/prometheus/promcfg.go

```diff
@@ -950,7 +953,7 @@ func (cg *ConfigGenerator) generateProbeConfig(
 		xc := labeler.GetRelabelingConfigs(m.TypeMeta, m.ObjectMeta, m.Spec.Targets.StaticConfig.RelabelConfigs)
 		relabelings = append(relabelings, generateRelabelConfig(xc)...)
 		cfg = append(cfg, yaml.MapItem{Key: "relabel_configs", Value: relabelings})
-	} else {
+	} else if m.Spec.Targets.Ingress != nil {
```

Comment:
suggestion: I would replace the `if () { ... } else if () {...}` structure by switch/case which I find more readable.

Reply:
I'm not seeing/understand how I can translate this `if else` block into a switch case, can you elaborate a bit?

Reply:
sure!

```golang
switch {
case m.Spec.Targets.StaticConfig != nil: // static targets defined
    ...
case m.Spec.Targets.Ingress != nil: // ingress targets defined
    ...
default:
    // The caller ensures that at least one of StaticConfig or Ingress is defined.
}

```